pkg/controller/health/reconciler.go

```diff
@@ -123,7 +123,8 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		} else {
 			var ok bool
 			if obj, ok = runtimeObject.(client.Object); !ok {
-				return ctrl.Result{}, fmt.Errorf("could not execute health check because object type %q is unsupported", ref.GroupVersionKind().String())
+				// do not requeue because there anyway will be another update event to fix the problem
+				return ctrl.Result{Requeue: false}, fmt.Errorf("could not execute health check because object type %q is unsupported", ref.GroupVersionKind().String())
```

Comment:
This will still requeue with exponential backoff, IISIC: see [doc string](https://github.com/kubernetes-sigs/controller-runtime/blob/a763c9a36c6f18660799384c9765348942bda53a/pkg/reconcile/reconcile.go#L90-L92) and [implementation](https://github.com/kubernetes-sigs/controller-runtime/blob/a763c9a36c6f18660799384c9765348942bda53a/pkg/internal/controller/controller.go#L297)

Reply:
Okay, sorry about that

Reply:
No worries :)